src/utils/v2/currency.ts

```diff
@@ -30,15 +30,15 @@ const CURRENCY_MAP: { [key in V1CurrencyOption]: V2CurrencyOption } = {
 export const V2CurrencyName = (
   currency?: V2CurrencyOption,
 ): CurrencyName | undefined =>
-  currency !== undefined ? V2_CURRENCY_METADATA[currency].name : undefined
+  currency !== undefined ? V2_CURRENCY_METADATA[currency]?.name : undefined
```

Comment:
@johnnyd-eth presumedly if `currency` was defined then `V2_CURRENCY_METADATA[currency]` is always defined. Any particular reason for this change?